opendcs-rest-api/src/main/java/org/opendcs/odcsapi/dao/ApiAuthorizationDAI.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *  Copyright 2024 OpenDCS Consortium and its Contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License")
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.opendcs.odcsapi.dao;
+
+import java.util.Set;
+
+import org.opendcs.odcsapi.sec.OpenDcsApiRoles;
+
+public interface ApiAuthorizationDAI extends ApiDaiBase
+{
+	Set<OpenDcsApiRoles> getRoles(String username) throws DbException;
+
+	String getUsernameForEdipi(long edipi) throws DbException;
```

Comment:
The authorization API should most likely just take the Request, or a plain string and return a `javax.security.Principal`, that leaves the details to the implementations.

Reply:
I really didn't want the DAO's to have to know about web service pieces in order to make them more flexible. I was able to refactor to move the edipi parsing internal though since it is definitely an implementation specific component.